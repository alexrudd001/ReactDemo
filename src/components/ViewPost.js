import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import ShowMessage from './ShowMessage';

const ViewPost = ({ match: { params: { id } } }) => {
    const [ post, setPost ] = useState(null);
    const [ messageType, setMessageType ] = useState('');

    useEffect(
        () => {
            fetch(`https://jsonplaceholder.typicode.com/posts/${id}`)
                .then((res) => res.json())
                .then((post) => {
                    setPost(post);
                })
                .catch((error) => {
                    setMessageType('error');
                });
        },
        [ id ]
    );
    return messageType === '' ? post ? (
        <div className="post">
            <div>
                <span style={idStyle}> Id : {post.id}</span>
                <h3> {post.title} </h3>
                <p> {post.body} </p>
                <Link
                    to={{
                        pathname: `/editPost/${id}`,
                        state: {
                            title: post.title,
                            body: post.body,
                            userId: post.userId
                        }
                    }}
                >
					Edit Post
                </Link>
                <span style={idStyle}> By User : {post.userId}</span>
            </div>
        </div>
    ) : (
        <p className="center">Loading...</p>
    ) : (
        <div className="center">
            <ShowMessage messageType={messageType} />
        </div>
    );
};
const idStyle = { float: 'right', fontSize: '0.8rem', fontWeight: 'bold' };

export default ViewPost;
