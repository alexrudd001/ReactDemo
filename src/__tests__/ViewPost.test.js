import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewPost from '../components/ViewPost';

test('renders ViewPost correctly', () => {
    const props = { match: { params: { id: 1 } } };
    const wrapper = shallow(<ViewPost {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
});
