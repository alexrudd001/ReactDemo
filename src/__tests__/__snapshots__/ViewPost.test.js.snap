// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ViewPost correctly 1`] = `
<p
  className="center"
>
  Loading...
</p>
`;
